[Bug]: Builds stopped working due to testcontainers failing on Jenkins

### Apache Hop version?

2.12-SNAPSHOT

### Java version?

17

### Operating system

Other

### What happened?

Builds stopped working due to testcontainers failing on Jenkins

```
[INFO] Running org.apache.hop.beam.transform.MergeJoinPipelineTest
Hop configuration file not found, not serializing: /home/jenkins/jenkins-agent/workspace/Hop_Hop_main/plugins/engines/beam/config/hop-config.json
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 32.73 s <<< FAILURE! -- in org.apache.hop.pipeline.transforms.cratedbbulkloader.http.CrateDBBulkImportTest
[ERROR] org.apache.hop.pipeline.transforms.cratedbbulkloader.http.CrateDBBulkImportTest -- Time elapsed: 32.73 s <<< ERROR!
org.testcontainers.containers.ContainerLaunchException: Container startup failed for image crate:latest
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)
	at org.testcontainers.junit.jupiter.TestcontainersExtension$StoreAdapter.start(TestcontainersExtension.java:280)
	at org.testcontainers.junit.jupiter.TestcontainersExtension$StoreAdapter.access$200(TestcontainersExtension.java:267)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.lambda$null$4(TestcontainersExtension.java:82)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.lambda$startContainers$5(TestcontainersExtension.java:82)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.startContainers(TestcontainersExtension.java:82)
	at org.testcontainers.junit.jupiter.TestcontainersExtension.beforeAll(TestcontainersExtension.java:56)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)
	... 9 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 10 more
```

### Issue Priority

Priority: 0

### Issue Component

Component: Infrastructure

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -21,7 +21,7 @@ def AGENT_LABEL = env.AGENT_LABEL ?: 'ubuntu'
 def JDK_NAME = env.JDK_NAME ?: 'jdk_17_latest'
 def MAVEN_NAME = env.MAVEN_NAME ?: 'maven_3_latest'
 
-def MAVEN_PARAMS = "-T 2 -U -B -e -fae -V -Dmaven.compiler.fork=true -Dsurefire.rerunFailingTestsCount=2"
+def MAVEN_PARAMS = "-T 2 -U -B -e -fae -V -Dmaven.compiler.fork=true -Dsurefire.rerunFailingTestsCount=2 -DSkipTestContainers=true"
 //removed fae (fail at end) by fn (fail never) to have an overview of the errors during development
 
 pipeline {

--- a/Jenkinsfile.daily
+++ b/Jenkinsfile.daily
@@ -21,7 +21,7 @@ def AGENT_LABEL = env.AGENT_LABEL ?: 'ubuntu'
 def JDK_NAME = env.JDK_NAME ?: 'jdk_17_latest'
 def MAVEN_NAME = env.MAVEN_NAME ?: 'maven_3_latest'
 
-def MAVEN_PARAMS = "-T 2 -U -B -e -fae -V -Dmaven.compiler.fork=true -Dsurefire.rerunFailingTestsCount=2"
+def MAVEN_PARAMS = "-T 2 -U -B -e -fae -V -Dmaven.compiler.fork=true -Dsurefire.rerunFailingTestsCount=2 -DSkipTestContainers=true"
 
 pipeline {
 

--- a/plugins/transforms/cratedbbulkloader/src/test/java/org/apache/hop/pipeline/transforms/cratedbbulkloader/http/BulkImportClientIT.java
+++ b/plugins/transforms/cratedbbulkloader/src/test/java/org/apache/hop/pipeline/transforms/cratedbbulkloader/http/BulkImportClientIT.java
@@ -17,8 +17,8 @@
 
 package org.apache.hop.pipeline.transforms.cratedbbulkloader.http;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import java.io.IOException;
 import java.sql.Connection;
@@ -28,50 +28,63 @@ import java.util.List;
 import org.apache.hop.core.exception.HopException;
 import org.apache.hop.pipeline.transforms.cratedbbulkloader.http.exceptions.CrateDBHopException;
 import org.apache.hop.pipeline.transforms.cratedbbulkloader.http.exceptions.UnauthorizedCrateDBAccessException;
-import org.junit.BeforeClass;
-import org.junit.ClassRule;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
 import org.testcontainers.cratedb.CrateDBContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
 import org.testcontainers.utility.MountableFile;
 
-public class BulkImportClientIT {
+@Testcontainers(disabledWithoutDocker = true)
+class BulkImportClientIT {
 
-  private static String CRATE_ENDPOINT;
-
-  private static Connection connection;
-
-  @ClassRule
   public static CrateDBContainer crateDBContainer =
       new CrateDBContainer("crate")
           .withCopyFileToContainer(
               MountableFile.forClasspathResource("crate.yml"), "/crate/config/crate.yml")
           .withExposedPorts(4200, 5432);
 
-  @BeforeClass
+  private static String crateEndpoint;
+  private static Connection connection;
+
+  @BeforeAll
   public static void setupAll() throws SQLException {
-    CRATE_ENDPOINT =
-        "http://"
-            + crateDBContainer.getHost()
-            + ":"
-            + crateDBContainer.getMappedPort(4200)
-            + "/_sql";
-    connection = crateDBContainer.createConnection("");
+    String skipTestContainers = System.getProperty("SkipTestContainers");
+    if (skipTestContainers == null) {
+      crateDBContainer.start();
 
-    connection
-        .createStatement()
-        .execute("CREATE TABLE crate.foo (id INT PRIMARY KEY, name VARCHAR(10), description TEXT)");
+      crateEndpoint =
+          "http://"
+              + crateDBContainer.getHost()
+              + ":"
+              + crateDBContainer.getMappedPort(4200)
+              + "/_sql";
+      connection = crateDBContainer.createConnection("");
 
-    connection.createStatement().execute("CREATE USER alice WITH (password='password')");
-    connection.createStatement().execute("GRANT ALL PRIVILEGES TO alice");
+      connection
+          .createStatement()
+          .execute(
+              "CREATE TABLE crate.foo (id INT PRIMARY KEY, name VARCHAR(10), description TEXT)");
 
-    connection.createStatement().execute("CREATE USER bob WITH (password='password')");
-    connection.createStatement().execute("GRANT DQL ON SCHEMA crate TO bob");
+      connection.createStatement().execute("CREATE USER alice WITH (password='password')");
+      connection.createStatement().execute("GRANT ALL PRIVILEGES TO alice");
+
+      connection.createStatement().execute("CREATE USER bob WITH (password='password')");
+      connection.createStatement().execute("GRANT DQL ON SCHEMA crate TO bob");
+    }
+  }
+
+  @AfterAll
+  public static void teardownAll() throws SQLException {
+    crateDBContainer.stop();
   }
 
   @Test
-  public void whenDataSizeGreaterThanMaxSize_shouldReturnRejectedRows()
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
+  void whenDataSizeGreaterThanMaxSize_shouldReturnRejectedRows()
       throws HopException, CrateDBHopException, IOException {
-    BulkImportClient client = new BulkImportClient(CRATE_ENDPOINT, "alice", "password");
+    BulkImportClient client = new BulkImportClient(crateEndpoint, "alice", "password");
 
     var response =
         client.batchInsert(
@@ -88,9 +101,10 @@ public class BulkImportClientIT {
   }
 
   @Test
-  public void whenRequestIsValid_shouldReturn200AndResult()
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
+  void whenRequestIsValid_shouldReturn200AndResult()
       throws HopException, SQLException, CrateDBHopException, IOException {
-    BulkImportClient client = new BulkImportClient(CRATE_ENDPOINT, "alice", "password");
+    BulkImportClient client = new BulkImportClient(crateEndpoint, "alice", "password");
     HttpBulkImportResponse response =
         client.batchInsert(
             "crate",
@@ -113,8 +127,9 @@ public class BulkImportClientIT {
   }
 
   @Test
-  public void whenWrongPassword_shouldThrowUnauthorizedException() {
-    BulkImportClient client = new BulkImportClient(CRATE_ENDPOINT, "alice", "wrongpassword");
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
+  void whenWrongPassword_shouldThrowUnauthorizedException() {
+    BulkImportClient client = new BulkImportClient(crateEndpoint, "alice", "wrongpassword");
 
     assertThrows(
         UnauthorizedCrateDBAccessException.class,
@@ -129,8 +144,9 @@ public class BulkImportClientIT {
   }
 
   @Test
-  public void whenWrongUser_shouldThrowUnauthorizedException() {
-    BulkImportClient client = new BulkImportClient(CRATE_ENDPOINT, "charlie", "apassword");
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
+  void whenWrongUser_shouldThrowUnauthorizedException() {
+    BulkImportClient client = new BulkImportClient(crateEndpoint, "charlie", "apassword");
 
     assertThrows(
         UnauthorizedCrateDBAccessException.class,
@@ -145,8 +161,9 @@ public class BulkImportClientIT {
   }
 
   @Test
-  public void whenUserNotAuthorized_shouldThrowUnauthorizedInsteadOfForbidden() {
-    BulkImportClient client = new BulkImportClient(CRATE_ENDPOINT, "bob", "password");
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
+  void whenUserNotAuthorized_shouldThrowUnauthorizedInsteadOfForbidden() {
+    BulkImportClient client = new BulkImportClient(crateEndpoint, "bob", "password");
 
     assertThrows(
         UnauthorizedCrateDBAccessException.class,

--- a/plugins/transforms/cratedbbulkloader/src/test/java/org/apache/hop/pipeline/transforms/cratedbbulkloader/http/CrateDBBulkImportTest.java
+++ b/plugins/transforms/cratedbbulkloader/src/test/java/org/apache/hop/pipeline/transforms/cratedbbulkloader/http/CrateDBBulkImportTest.java
@@ -51,6 +51,7 @@ import org.apache.hop.pipeline.transforms.cratedbbulkloader.CrateDBBulkLoaderMet
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
@@ -60,52 +61,90 @@ import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
 @Testcontainers(disabledWithoutDocker = true)
-public class CrateDBBulkImportTest {
+class CrateDBBulkImportTest {
 
-  @Container
-  public static CrateDBContainer crateDBContainer = new CrateDBContainer("crate").withReuse(true);
+  @Container public static CrateDBContainer crateDBContainer = new CrateDBContainer("crate");
 
   private static Connection connection;
 
   @BeforeAll
-  public static void init() throws Exception {
-    HopClientEnvironment.init(
-        List.of(
-            DatabasePluginType.getInstance(),
-            TwoWayPasswordEncoderPluginType.getInstance(),
-            ValueMetaPluginType.getInstance()));
-    HopEnvironment.init();
-    HopLogStore.init(true, true);
-    connection = crateDBContainer.createConnection("");
-    String createTableQuery =
-        """
-                CREATE TABLE doc.person
-                (
-                  id INTEGER PRIMARY KEY,
-                  name TEXT,
-                  score DOUBLE PRECISION)
-                ;
-                """;
-    executeUpdate("DROP TABLE IF EXISTS doc.person");
-    executeUpdate("DROP USER IF EXISTS bob");
-    executeUpdate(createTableQuery);
-    executeUpdate("REFRESH TABLE doc.person");
-    executeUpdate("CREATE USER bob WITH PASSWORD 'password'");
-    executeUpdate("GRANT DML ON SCHEMA doc TO bob");
-  }
-
-  @BeforeEach
-  public void setup() throws Exception {
-    executeUpdate("DELETE FROM doc.person");
+  static void init() throws Exception {
+    String skipTestContainers = System.getProperty("SkipTestContainers");
+    if (skipTestContainers == null) {
+      HopClientEnvironment.init(
+          List.of(
+              DatabasePluginType.getInstance(),
+              TwoWayPasswordEncoderPluginType.getInstance(),
+              ValueMetaPluginType.getInstance()));
+      HopEnvironment.init();
+      HopLogStore.init(true, true);
+      connection = crateDBContainer.createConnection("");
+      String createTableQuery =
+          """
+                            CREATE TABLE doc.person
+                            (
+                              id INTEGER PRIMARY KEY,
+                              name TEXT,
+                              score DOUBLE PRECISION)
+                            ;
+                            """;
+      executeUpdate("DROP TABLE IF EXISTS doc.person");
+      executeUpdate("DROP USER IF EXISTS bob");
+      executeUpdate(createTableQuery);
+      executeUpdate("REFRESH TABLE doc.person");
+      executeUpdate("CREATE USER bob WITH PASSWORD 'password'");
+      executeUpdate("GRANT DML ON SCHEMA doc TO bob");
+    }
   }
 
   @AfterAll
-  public static void shutdown() throws Exception {
-    executeUpdate("DROP TABLE doc.person;");
+  static void shutdown() throws Exception {
+    String skipTestContainers = System.getProperty("SkipTestContainers");
+    if (skipTestContainers == null) {
+      executeUpdate("DROP TABLE doc.person;");
+    }
+  }
+
+  private static int executeUpdate(String query) throws Exception {
+    try (Statement statement = connection.createStatement()) {
+      return statement.executeUpdate(query);
+    }
+  }
+
+  private static ResultSet executeQuery(String query) throws Exception {
+    Statement statement = connection.createStatement();
+    return statement.executeQuery(query);
+  }
+
+  private static Stream<Arguments> getBatchSize() {
+    return Stream.of(
+        Arguments.of(1),
+        Arguments.of(2),
+        Arguments.of(3),
+        Arguments.of(4),
+        Arguments.of(5),
+        Arguments.of(6),
+        Arguments.of(7),
+        Arguments.of(8),
+        Arguments.of(9),
+        Arguments.of(10),
+        Arguments.of(20),
+        Arguments.of(30),
+        Arguments.of(40),
+        Arguments.of(50));
+  }
+
+  @BeforeEach
+  void setup() throws Exception {
+    String skipTestContainers = System.getProperty("SkipTestContainers");
+    if (skipTestContainers == null) {
+      executeUpdate("DELETE FROM doc.person");
+    }
   }
 
   @ParameterizedTest
   @MethodSource("getBatchSize")
+  @DisabledIfSystemProperty(named = "SkipTestContainers", matches = "true")
   void given_batch_size__when_http_insert__should_persist_all_items(Integer batchSize)
       throws Exception {
     Mockito.mock(Pipeline.class);
@@ -185,34 +224,5 @@ public class CrateDBBulkImportTest {
     final int resultSize = resultSet.getInt("c");
 
     assertEquals(5, resultSize);
-  }
-
-  private static int executeUpdate(String query) throws Exception {
-    try (Statement statement = connection.createStatement()) {
-      return statement.executeUpdate(query);
-    }
-  }
-
-  private static ResultSet executeQuery(String query) throws Exception {
-    Statement statement = connection.createStatement();
-    return statement.executeQuery(query);
-  }
-
-  private static Stream<Arguments> getBatchSize() {
-    return Stream.of(
-        Arguments.of(1),
-        Arguments.of(2),
-        Arguments.of(3),
-        Arguments.of(4),
-        Arguments.of(5),
-        Arguments.of(6),
-        Arguments.of(7),
-        Arguments.of(8),
-        Arguments.of(9),
-        Arguments.of(10),
-        Arguments.of(20),
-        Arguments.of(30),
-        Arguments.of(40),
-        Arguments.of(50));
   }
 }

